archinstall/lib/crypt.py

```diff
@@ -69,3 +75,52 @@ def crypt_yescrypt(plaintext: str) -> str:
 		raise ValueError('crypt() returned NULL')
 
 	return crypt_hash.decode('utf-8')
+
+
+def _get_fernet(salt: bytes, password: Password) -> Fernet:
+	# https://cryptography.io/en/latest/hazmat/primitives/key-derivation-functions/#argon2id
+	kdf = Argon2id(
+		salt=salt,
+		length=32,
+		iterations=1,
+		lanes=4,
+		memory_cost=64 * 1024,
+		ad=None,
+		secret=None,
+	)
+
+	key = base64.urlsafe_b64encode(
+		kdf.derive(
+			password.plaintext.encode('utf-8')
```

Comment:
`_get_fernet()`, used by `encrypt()` and `decrypt()`, only uses `Password` for plaintext.